### PR TITLE
Changes swarmers to need nearby pylons to build.

### PR DESCRIPTION
**Pull request by lordpidey**
## About The Pull Request

Nerfs swarmers by weakly tying them to a limited area that they can expand.

Swarmers will only be able to deconstruct/absorb/build when near a pylon, and swarmers can build new pylons while in the range of an existing pylon.  (The first pylon is an exception, and can be placed anywhere, for no resources)

## Why It's Good For The Game

Swarmers are pretty much a roundender.  Once they're on the station, they're impossible to get rid of.  This will give the crew a way to actually END the threat by destroying all of the pylons.  Once all pylons are destroyed, the swarmers will have no way to absorb resources, and thus no way to procreate.

## Changelog
:cl:
balance: Swarmers can now only operate in areas effected by a swarmer pylon.
add: Swarmers can now construct pylons for 25 resources.
/:cl:

### CODE REVIEW

File: code/_onclick/hud/swarmer.dm [hunk 1]
```diff
@@ -13,6 +13,22 @@
...
+/**
+  * # SwarmerPylonButton
+  *
+  * The ui object operating as a button to create pylons.
+  */
+
+/obj/screen/swarmer/Pylon
+	icon_state = "ui_barricade"
+	name = "Create pylon (Costs 5 Resources)"
+	desc = "Creates a pylon, which expands the area swarmers may operate in. (Costs 25 resources)"
```
Comment (ArcaneMusic) [inline code/_onclick/hud/swarmer.dm:25]: So, is this intended to be 25 or 5? Further down it's set to 25 so the name could just be a typo.

Comment (lordpidey) [reply code/_onclick/hud/swarmer.dm:25]: My intention was 25, typing 5 was a mistake.

This value was chosen via guesstimate.  I hold no attachment to this value whatsoever.

**Comment by nianjiilical** [comment]
the basic idea of tying swarmers to specific constructs is a good idea, but on paper completely removing their ability to absorb/deconstruct without a pylon feels like too harsh a nerf, especially when they actually need those resources to make pylons to begin with

i think some slightly more balanced tweaks could be any combination of the following:

a) massively nerf swarmers' deconstruction speed when not near pylons, so they can still eat without one, but still need pylons to make any real ground
b) give pylon creation a large cooldown per swarmer, so that small numbers can't effectively claim large areas, but their ability to hold territory expands with their numbers
c) make pylons free but only allow each swarmer to support one, deconstructing it manually refunds the cooldown but if a pylon is destroyed that swarmer cant make a new one for a while, so that swarmers absolutely need to multiply to spread and killing a pylon is a meaningful blow to the colony

it feels like the tricky part is hitting that balance between "crewmembers killing pylons is a significant blow to swarmers", "a single well-played swarmer can rebuild even if they're alone and resourceless", and "swarmers are tied to pylons in a way that they become stronger the more of them there are"

**Comment by Shadowflame909** [comment]
I don't like the fact that "making everything like blob" has become the end-goal of simple mob group balance in recent days.

Whats next, killing the midwife spider will kill all the spiders? Killing the xeno queen causes all the xenomorphs to spontaneously gib?

**Comment by lordpidey** [comment]
@nianjiilical , I kind of did opt for #2, by making pylons have a decently high resource cost to create  (25)

**Comment by lordpidey** [comment]
> killing the midwife spider will kill all the spiders? Killing the xeno queen causes all the xenomorphs to spontaneously gib?

Well, for one, it's not just ONE target here that needs to be killed.  There's going to be a fair number of pylons.  It's not a single point of failure, its a redundant network that needs to be dismantled, and one that can be repaired if damaged.

Second, destroying the network does NOT destroy the swarmers, it's not like in your comparison where they gib/die.

**Comment by Shadowflame909** [comment]
> Second, destroying the network does NOT destroy the swarmers, it's not like in your comparison where they gib/die.

>Swarmers will only be able to deconstruct/absorb/build when near a pylon, and swarmers can build new pylons while in the range of an existing pylon.

I mean...it's basically blob but easier since all the warmers are non-lethal and have very shitty defenses that can be broken with two hits of anything a borg has on them.

Nerf is brutal and reductive to the antagonist is all I'm saying, because its making it more like another antag without any of their strengths.

You build a Pylon near the ORE Silo to destroy it and cripple the station, now you have to mass enough resources to build another Pylon to actually be able to destroy anything else within range of your previous pylon,, the 60 number of crews wandering the halls see you, and you have to protect your pylon, be able to replicate more warmers and another pylon, and essentially you get the most uber nerfed and overwhelmed antag I've seen in a long time.

The swarmers are getting swarmed with multiple objectives, no way of doing them all in an actual decent manner that wont get them exposed. (Because they're forced to stay at one area if they want to replicate/destroy like at all.) and quite literally you'd only need one guy to find a swarmer anywhere, have them ignore the swarmer/two shot it. And just start hitting the pylon because the swarmer is no no longer an issue at all, it's merely the worst defender in the game.

**Comment by lordpidey** [comment]
Closing this, it has been made apparent that other swarmer related PRs are preferred over this one.

**Pull request closed without merging**